src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -0,0 +1,317 @@
+  private final Version version;
+
+  @VisibleForTesting
+  public JestElasticsearchClient(String address) {
+    try {
+      JestClientFactory factory = new JestClientFactory();
+      factory.setHttpClientConfig(new HttpClientConfig.Builder(address)
+          .multiThreaded(true)
+          .build()
+      );
+      this.client = factory.getObject();
+      this.version = getServerVersion();
+    } catch (IOException e) {
+      throw new ConnectException(
+          "Couldn't start ElasticsearchSinkTask due to connection error:",
+          e
+      );
+    } catch (ConfigException e) {
+      throw new ConnectException(
+          "Couldn't start ElasticsearchSinkTask due to configuration error:",
+          e
+      );
+    }
+  }
+
+  public JestElasticsearchClient(Map<String, String> props) {
+    try {
+      ElasticsearchSinkConnectorConfig config = new ElasticsearchSinkConnectorConfig(props);
+      final int connTimeout = config.getInt(
+              ElasticsearchSinkConnectorConfig.CONNECTION_TIMEOUT_MS_CONFIG);
+      final int readTimeout = config.getInt(
+              ElasticsearchSinkConnectorConfig.READ_TIMEOUT_MS_CONFIG);
+
+      List<String> address =
+              config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
+      JestClientFactory factory = new JestClientFactory();
+      factory.setHttpClientConfig(new HttpClientConfig.Builder(address)
+              .connTimeout(connTimeout)
+              .readTimeout(readTimeout)
+              .multiThreaded(true)
+              .build()
+      );
+      this.client = factory.getObject();
+      this.version = getServerVersion();
+    } catch (IOException e) {
+      throw new ConnectException(
+              "Couldn't start ElasticsearchSinkTask due to connection error:",
+              e
+      );
+    } catch (ConfigException e) {
+      throw new ConnectException(
+              "Couldn't start ElasticsearchSinkTask due to configuration error:",
+              e
+      );
+    }
+  }
+
+  private Version getServerVersion() throws IOException {
+    // Default to newest version for forward compatibility
+    Version defaultVersion = Version.SIX;
+
+    NodesInfo info = new NodesInfo.Builder().addCleanApiParameter("version").build();
+    JsonObject result = this.client.execute(info).getJsonObject();
+    if (result == null) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    }
+
+    JsonObject nodesRoot = result.get("nodes").getAsJsonObject();
+    if (nodesRoot == null || nodesRoot.entrySet().size() == 0) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    }
+
+    JsonObject nodeRoot = nodesRoot.entrySet().iterator().next().getValue().getAsJsonObject();
+    if (nodeRoot == null) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    }
+
+    String esVersion = nodeRoot.get("version").getAsString();
```

Comment:
Does this *should* work for all of the ES versions and the above logic is really handling exceptional cases? If so, then perhaps this method should have doc that explains that. And, the log messages should maybe be a bit more descriptive so that we can more easily tell which path the logic takes.

Reply:
Yes, it should work for all versions.  I'll make the errors more descriptive.